test/controllers/beneficial.owner.other.controller.spec.ts

```diff
@@ -188,6 +209,40 @@ describe("BENEFICIAL OWNER OTHER controller", () => {
       expect(resp.text).toContain(ErrorMessages.COUNTY_STATE_PROVINCE_REGION_INVALID_CHARACTERS);
       expect(resp.text).toContain(ErrorMessages.POSTCODE_ZIPCODE_INVALID_CHARACTERS);
     });
+
+    test(`Service address from the ${BENEFICIAL_OWNER_OTHER_PAGE} is present when same address is set to no`, async () => {
+      mockPrepareData.mockImplementation( () => BENEFICIAL_OWNER_OTHER_OBJECT_MOCK_WITH_SERVICE_ADDRESS_NO);
+      await request(app).post(BENEFICIAL_OWNER_OTHER_URL);
+      expect(mapFieldsToDataObject).toHaveBeenCalledWith({}, ServiceAddressKeys, AddressKeys);
+      const data: ApplicationDataType = mockSetApplicationData.mock.calls[0][1];
+      expect(data[ServiceAddressKey]).toEqual(DUMMY_DATA_OBJECT);
+    });
+
+    test(`Service address from the ${BENEFICIAL_OWNER_OTHER_PAGE} is empty when same address is set to yes`, async () => {
+      mockPrepareData.mockImplementation( () => BENEFICIAL_OWNER_OTHER_OBJECT_MOCK_WITH_SERVICE_ADDRESS_YES);
+      mockSetApplicationData.mockImplementation( () => setApplicationData);
```

Comment:
the lines `mockSetApplicationData.mockImplementation( () => setApplicationData);` can be removed. 
setApplicationData is actually the same mock function as mockSetApplicationData. Jest will mock the setApplicationData when jest.mock('../../src/utils/application.data'); runs, but in order for us to access some of the Mock features we just create an alias to it by doing the ".... as jest.Mock". Then we can access the .mock.calls etc..

Reply:
it's not a showstopper so i'll approve, but if you get time, the `mockSetApplicationData.mockImplementation( () => setApplicationData);` lines can be removed, the test still passes with out them.